ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/bytes/ByteUtils.kt

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.quic.bytes
+
+import io.ktor.utils.io.core.*
+
+/**
+ * Reads an unsigned byte from a [ByteReadPacket]. Returns null if EOF
+ */
+@ExperimentalUnsignedTypes
+internal fun ByteReadPacket.readUByteOrNull() = if (isNotEmpty) readUByte() else null
```

Comment:
It would be better if required amount of bytes will be checked before all read calls

Reply:
what do you think about this?
```
@ExperimentalUnsignedTypes
internal inline fun ByteReadPacket.readUByteOrElse(elseBlock: () -> Nothing): UByte {
    if (isEmpty) {
        elseBlock()
    }
    return readUByte()
} 
``` 
```
val ubyte = readUByteOrElse { return null }
```

Reply:
from what I see in code, may be even just calling `isEmpty` in place will be better, for readability at least as there is not a lot of calls to this function comparing to `readVarInt

Reply:
I think for consistency it would be better to  make it the same as `readVarIntOrElse`, as where are a lot of usages of the last one and it wont be a problem for reader to understand the meaning quickly. Besides that, i will also use it to read packet headers, so there will be more occurrences of this function